diff --git a/nodereferrer.module b/nodereferrer.module
old mode 100644
new mode 100755
index c41f1ae..c4381a5
--- a/nodereferrer.module
+++ b/nodereferrer.module
@@ -479,7 +479,10 @@ function nodereferrer_field_formatter_view($entity_type, $entity, $field, $insta
     case 'nodereferrer_default':
     case 'nodereferrer_plain':
       foreach ($items as $delta => $item) {
-        $nodes = node_load_multiple(array_keys($item['items']));
+        $nodes = array();
+        if (is_array($item['items'])) {
+          $nodes = node_load_multiple(array_keys($item['items']));
+        }
         $rendered_nodes = array();
         foreach ($nodes as $node) {
           if ($display['type'] == 'nodereferrer_default') {
