diff --git a/comment_goodness.module b/comment_goodness.module
index ffad694..fd6cbce 100644
--- a/comment_goodness.module
+++ b/comment_goodness.module
@@ -204,7 +204,7 @@ function comment_goodness_permalink($cid) {
     // Return the node view, this will show the correct comment in context.
     return menu_execute_active_handler('node/' . $node->nid, FALSE);
   }
-  drupal_not_found();
+  return MENU_NOT_FOUND;
 }
 
 /**
