diff --git a/comment_revisions.module b/comment_revisions.module
index db65149..92fce89 100644
--- a/comment_revisions.module
+++ b/comment_revisions.module
@@ -10,7 +10,7 @@
 function comment_revisions_views_api() {
   return array(
     'api' => 3,
-    'path' => drupal_get_path('module', 'comment_revision') . '/views',
+    'path' => drupal_get_path('module', 'comment_revisions') . '/views',
   );
 }
 
@@ -237,7 +237,7 @@ function comment_revisions_entity_info_alter(&$entity_info) {
   $entity_info['comment']['revision table'] = 'comment_revision';
   $entity_info['comment']['entity keys']['revision'] = 'vid';
   $entity_info['comment']['controller class'] = 'CommentRevisionController';
-  module_load_install('comment_revision');
+  module_load_include('install', 'comment_revisions');
   $schema = comment_revisions_schema();
   foreach ($schema['comment_revision']['fields'] as $field => $data) {
     $entity_info['comment']['schema_fields_sql']['revision table'][] = $field;
