diff --git a/freelinking.utilities.inc b/freelinking.utilities.inc
index 3d51237..b96ee21 100644
--- a/freelinking.utilities.inc
+++ b/freelinking.utilities.inc
@@ -84,6 +84,7 @@ function freelinking_get_plugins($format = 'all', $reset = FALSE) {
 
   if (empty($plugins[$format]) || $reset) {
     $freelinking = module_invoke_all('freelinking');
+    drupal_alter('freelinking', $freelinking);
 
     // Validate & standardize plugins.
     foreach ($freelinking as $name => &$plugin) {
